test/Microsoft.AspNetCore.Razor.Language.Test/Legacy/HtmlMarkupParserTests.cs

```diff
@@ -0,0 +1,206 @@
+        {
+            // Arrange
+            var dashSymbol = new HtmlSymbol("-", HtmlSymbolType.Text);
+
+            // Act & Assert
+            Assert.True(HtmlMarkupParser.IsDashSymbol(dashSymbol));
+        }
+
+        [Fact]
+        public void AcceptAllButLastDoubleHypens_ReturnsTheOnlyDoubleHyphenSymbol()
+        {
+            // Arrange
+            var sut = CreateTestParserForContent("-->");
+
+            // Act
+            var symbol = sut.AcceptAllButLastDoubleHypens();
+
+            // Assert
+            Assert.Equal(doubleHyphenSymbol, symbol);
+            Assert.True(sut.At(HtmlSymbolType.CloseAngle));
+            Assert.Equal(doubleHyphenSymbol, sut.PreviousSymbol);
+        }
+
+        [Fact]
+        public void AcceptAllButLastDoubleHypens_ReturnsTheDoubleHyphenSymbolAfterAcceptingTheDash()
+        {
+            // Arrange
+            var sut = CreateTestParserForContent("--->");
+
+            // Act
+            var symbol = sut.AcceptAllButLastDoubleHypens();
+
+            // Assert
+            Assert.Equal(doubleHyphenSymbol, symbol);
+            Assert.True(sut.At(HtmlSymbolType.CloseAngle));
+            Assert.True(HtmlMarkupParser.IsDashSymbol(sut.PreviousSymbol));
+        }
+
+        [Fact]
+        public void IsHtmlCommentAhead_ReturnsTrueForEmptyCommentTag()
+        {
+            // Arrange
+            var sut = CreateTestParserForContent("---->");
+
+            // Act & Assert
+            Assert.True(sut.IsHtmlCommentAhead());
+        }
+
+        [Fact]
+        public void IsHtmlCommentAhead_ReturnsTrueForValidCommentTag()
+        {
+            // Arrange
+            var sut = CreateTestParserForContent("-- Some comment content in here -->");
+
+            // Act & Assert
+            Assert.True(sut.IsHtmlCommentAhead());
+        }
+
+        [Fact]
+        public void IsHtmlCommentAhead_ReturnsTrueForValidCommentTagWithExtraDashesAtClosingTag()
+        {
+            // Arrange
+            var sut = CreateTestParserForContent("-- Some comment content in here ----->");
+
+            // Act & Assert
+            Assert.True(sut.IsHtmlCommentAhead());
+        }
+
+        [Fact]
+        public void IsHtmlCommentAhead_ReturnsTrueForValidCommentTagWithExtraInfoAfter()
+        {
+            // Arrange
+            var sut = CreateTestParserForContent("-- comment --> the first part is a valid comment without the Open angle and bang symbols");
+
+            // Act & Assert
+            Assert.True(sut.IsHtmlCommentAhead());
+        }
+
+        [Fact]
+        public void IsHtmlCommentAhead_ReturnsFalseForNotClosedComment()
+        {
+            // Arrange
+            var sut = CreateTestParserForContent("-- not closed comment");
+
+            // Act & Assert
+            Assert.False(sut.IsHtmlCommentAhead());
+        }
+
+        [Fact]
+        public void IsHtmlCommentAhead_ReturnsFalseForCommentWithoutLastClosingAngle()
```

Comment:
Add a case for `@DateTime.Now` inside the comment.

Reply:
Done